Ataque do jogador

Marcos-Fidelis opened the pull request:
eu adicionei o brutão do ataque e fiz o hp do primeiro monstro deminuir, ainda n faz nada, mas agt vai usar isso pra fazer eles sumirem

Marcos-Fidelis committed:
novamente, ainda ta brutão

eu fiz a hp do monstro aparecer e eu fiz o brutão do ataque. qnd a hitbox do ataque colide com o monstro aparece um circulo branco atras dele. agora falta agt fazer aquele plano de fazer um array de monstros pra poder fazer o teste com mais de um

Marcos-Fidelis committed:
Merge branch 'ataque-jogador' of https://github.com/VelocityIsNotSpeed/Grupo42-IntProg1 into ataque-jogador

Marcos-Fidelis committed:
agora faz dano

eu mexi nele dnv e agr ele toma dano, o dnao n faz nada ainda, mas ele toma dano ¯\_(ツ)_/¯
qnd agt fizer o array com varios monstros  é só setar um if pra qnd a vida dele chegar a zero mudar o  bool "seguindo" e colocar um if em desenhar pra if(seguindo) q ele some qnd zerar a vida dele

VelocityIsNotSpeed commented:
Essa PR implementa #27

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -43,7 +45,11 @@ int main(void)
     gs->inim.rot = 0;
     gs->inim.hp = 120;
-    gs->inim.seguindo = false;
+    gs->inim.seguindo = true;
```
Não é pra usar `seguindo`.

VelocityIsntSpeed replied:
Consertei com o commit 436becb

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -43,7 +45,11 @@ int main(void)
     gs->inim.hp = 120;
-    gs->inim.seguindo = false;
+    gs->inim.seguindo = true;
+    gs->inim.hpos.x = gs->inim.pos.x-12;
```
```suggestion
    gs->inim.hpos.x = gs->inim.pos.x - 12;
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -43,7 +45,11 @@ int main(void)
-    gs->inim.seguindo = false;
+    gs->inim.seguindo = true;
+    gs->inim.hpos.x = gs->inim.pos.x-12;
+    gs->inim.hpos.y = gs->inim.pos.y+22;
```
Faltando espaços.
```suggestion
    gs->inim.hpos.y = gs->inim.pos.y + 22;
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -76,6 +81,16 @@ int main(void)
         gs->cam.offset.x += GetScreenWidth() / 2;
         gs->cam.offset.y += GetScreenHeight() / 2;
         gs->cam.target = gs->jog.pos;
+        // Atualizar hp do inimigo
```
Era pra ter uma linha vazia separando isso do código de cima
```suggestion

        // Atualizar hp do inimigo
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -32,6 +32,8 @@ int main(void)
     /// Sprite do jogador.
     const Texture2D SPRITE_JOG = LoadTexture("tex/protag.png");
+    gs->jog.hitbox = gs->jog.pos;
+    gs->jog.hitbox.x = gs->jog.hitbox.x-100;
```
```suggestion
    gs->jog.hitbox.x = gs->jog.hitbox.x - 100;
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -64,7 +70,6 @@ int main(void)
 
         // Rotacao do jogador
         gs->jog.rot = Vector2Angle(gs->jog.pos, PosWorldDoCursor(gs));
-
```
Não era pra ter deletado essa linha

VelocityIsntSpeed replied:
Consertei com o commit afd3a6d

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -76,6 +81,16 @@ int main(void)
         gs->cam.offset.y += GetScreenHeight() / 2;
         gs->cam.target = gs->jog.pos;
+        // Atualizar hp do inimigo
+        if( IsMouseButtonPressed(MOUSE_LEFT_BUTTON))
```
```suggestion
        if (IsMouseButtonPressed(MOUSE_LEFT_BUTTON))
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -76,6 +81,16 @@ int main(void)
+        // Atualizar hp do inimigo
+        if( IsMouseButtonPressed(MOUSE_LEFT_BUTTON))
+        {
+            if(CheckCollisionCircles(gs->jog.hitbox, 10,gs->inim.pos, 20))
```
```suggestion
            if (CheckCollisionCircles(gs->jog.hitbox, 10, gs->inim.pos, 20))
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -76,6 +81,16 @@ int main(void)
+        {
+            if(CheckCollisionCircles(gs->jog.hitbox, 10,gs->inim.pos, 20))
+            {
+                gs->inim.hp = gs->inim.hp - 20;
```
Assim é mais claro:
```suggestion
                gs->inim.hp -= 20;
```

VelocityIsntSpeed commented on jogo42-cb/jogador.c:
```diff
@@ -75,6 +75,10 @@ void MoverJog(GameState* gs)
     }
+    // Ataque
+    gs->jog.hitbox = gs->jog.pos;
+    gs->jog.hitbox.x = gs->jog.hitbox.x + 60*cos(gs->jog.rot/60);
```
60 graus não é igual a 1 radiano. O valor certo é aproximadamente `57.2958`.

Mas também não é pra dividir por esse valor. Em `raymath.h` está definida a constante `DEG2RAD`, que tem o valor tal que quando multiplicado por um ângulo em graus, o resultado é esse mesmo ângulo em radianos (esse valor é `PI / 180.0f`).

Obs: Existe também a constante `RAD2DEG`, para fazer a conversão inversa. O `2` nesses nomes representam a palavra inglesa "to".

Trecho relevante de `raymath.h`:
```c
/* ... */

//----------------------------------------------------------------------------------
// Some basic Defines
//----------------------------------------------------------------------------------
#ifndef PI
    #define PI 3.14159265358979323846f
#endif

#define DEG2RAD (PI/180.0f)
#define RAD2DEG (180.0f/PI)

/* ... */
```

```suggestion
    gs->jog.hitbox.x = gs->jog.hitbox.x + RAD2DEG * cos(gs->jog.rot * DEG2RAD);
```

VelocityIsntSpeed replied:
Consertei com commit 9523dee

VelocityIsntSpeed commented on jogo42-cb/jogador.c:
```diff
@@ -75,6 +75,10 @@ void MoverJog(GameState* gs)
+    // Ataque
+    gs->jog.hitbox = gs->jog.pos;
+    gs->jog.hitbox.x = gs->jog.hitbox.x + 60*cos(gs->jog.rot/60);
+    gs->jog.hitbox.y = gs->jog.hitbox.y + 60*sin(gs->jog.rot/60);
```
Ver comentário da linha acima.
```suggestion
    gs->jog.hitbox.y = gs->jog.hitbox.y + RAD2DEG * sin(gs->jog.rot * DEG2RAD);
```

VelocityIsntSpeed replied:
Consertei com commit 9523dee

VelocityIsntSpeed commented on jogo42-cb/desenhar.c:
```diff
@@ -111,12 +111,23 @@ void Desenhar(const GameState* gs, const Texture2D* spriteJog)
+
         // Jogador
         DesenharJogador(gs, spriteJog);
+        if( IsMouseButtonDown(MOUSE_LEFT_BUTTON))
```
```suggestion
        if (IsMouseButtonDown(MOUSE_LEFT_BUTTON))
```

VelocityIsntSpeed commented on jogo42-cb/desenhar.c:
```diff
@@ -111,12 +111,23 @@ void Desenhar(const GameState* gs, const Texture2D* spriteJog)
         DesenharJogador(gs, spriteJog);
+        if( IsMouseButtonDown(MOUSE_LEFT_BUTTON))
+        {
+            if(CheckCollisionCircles(gs->jog.hitbox, 10,gs->inim.pos, 20))
```
```suggestion
            if(CheckCollisionCircles(gs->jog.hitbox, 10, gs->inim.pos, 20))
```

VelocityIsntSpeed commented on jogo42-cb/desenhar.c:
```diff
@@ -111,12 +111,23 @@ void Desenhar(const GameState* gs, const Texture2D* spriteJog)
+        if(gs->inim.seguindo)
+        {
+            DrawCircleV(gs->inim.pos, 20, (Color){223, 0, 0, 255});
+            DrawText(gs->inim.vida,gs->inim.hpos.x, gs->inim.hpos.y, 20, RED );
```
```suggestion
            DrawText(gs->inim.vida, gs->inim.hpos.x, gs->inim.hpos.y, 20, RED);
```

VelocityIsntSpeed commented on jogo42-cb/desenhar.c:
```diff
@@ -111,12 +111,23 @@ void Desenhar(const GameState* gs, const Texture2D* spriteJog)
         // Inimigo
-        DrawCircleV(gs->inim.pos, 20, (Color){223, 0, 0, 255});
-
+        if(gs->inim.seguindo)
```
Não é pra usar seguindo.

VelocityIsntSpeed replied:
Consertei com 436becb

VelocityIsntSpeed reviewed (changes requested):
Tem mais depois.

VelocityIsntSpeed committed:
Aplica algumas sugestões da review

relacionadas à estilo.

VelocityIsntSpeed committed:
Readiciona linha vazia que tinha sido incorretamente removida

De acordo com o review a PR #31

VelocityIsntSpeed committed:
Conserta conversão entre graus e radianos

De acordo com a review da PR #31

VelocityIsntSpeed committed:
Remove o uso de Inimigo.seguindo

De acordo com a review da PR #31

VelocityIsntSpeed committed:
Remove círculo branco indicador de acerto

Com o hp do inimigo diminuindo, é desnecessário. E de qualquer forma, esse não era um bom jeito de implementar isso.

VelocityIsntSpeed committed:
Refatora código de ataque

- Remove código de MoverJog
- Remove membro Jog.hitbox
- Coloca o código de ataque na função AtaqueJogador
- Adiciona constantes JOG_ATQ_DIST e JOG_ATQ_RAIO

VelocityIsntSpeed committed:
Refatora desenhar do texto do hp do inimigo

VelocityIsntSpeed committed:
Adc. funcao Vector2AndarAte

VelocityIsntSpeed committed:
Remove núm. mágico do dano de ataque do jogador

VelocityIsntSpeed committed:
Remove include de stdlib.h

Não estava sendo usado

VelocityIsntSpeed committed:
Adc. indicador gráfico de ataque do jogador

VelocityIsntSpeed committed:
Merge branch 'master' into ataque-jogador

VelocityIsntSpeed committed:
Merge branch 'master' into ataque-jogador
